python/packages/autogen-ext/src/autogen_ext/models/_openai/_openai_client.py

```diff
@@ -636,13 +638,29 @@ async def create_stream(
         full_tool_calls: Dict[int, FunctionCall] = {}
         completion_tokens = 0
         logprobs: Optional[List[ChatCompletionTokenLogprob]] = None
+        empty_chunk_count = 0
+
         while True:
             try:
                 chunk_future = asyncio.ensure_future(anext(stream))
                 if cancellation_token is not None:
                     cancellation_token.link_future(chunk_future)
                 chunk = await chunk_future
 
+                # This is to address a bug in AzureOpenAIChatCompletionClient. OpenAIChatCompletionClient works fine.
+                #  https://github.com/microsoft/autogen/issues/4213
+                if len(chunk.choices) == 0:
```

Comment:
Do the chunks with no choices have other data set on other fields of the delta?

Reply:
I can't see anything int the `chunk.__dict__`. In the consecutive chunks, the choices gets populated properly and has all the expected fields.
```
{'id': '', 'choices': [], 'created': 0, 'model': '', 'object': '', 'service_tier': None, 'system_fingerprint': None, 'usage': None}
```